### PR TITLE
Introduced a registry for LB policies.

### PR DESCRIPTION
**Pull request by dgquintas**
In preparation for the new round robin lb policy.

### CODE REVIEW

**Comment by dgquintas** [comment]
Green-ish

File: src/core/client_config/lb_policy_factory.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+struct grpc_lb_policy_factory_vtable {
+  void (*ref)(grpc_lb_policy_factory *factory);
+  void (*unref)(grpc_lb_policy_factory *factory);
+
+  /** Implementation of grpc_lb_policy_factory_create_lb_policy */
+  grpc_lb_policy *(*create_lb_policy)(grpc_lb_policy_factory *factory,
+                                      grpc_subchannel **subchannels,
+                                      size_t num_subchannels);
```
Comment (ctiller) [inline src/core/client_config/lb_policy_factory.h]: Can we remove these two arguments (subchannels, num_subchannels), and replace them with a grpc_lb_policy_args *args... where:

```
struct grpc_lb_policy_args {
  grpc_lb_policy_factory *factory;
  grpc_subchannel **subchannels;
};
```

Rationale: it'll make extending this argument list so much easier in the future

Comment (dgquintas) [inline src/core/client_config/lb_policy_factory.h]: I guess you meant 

```
struct grpc_lb_policy_args {
  grpc_subchannel **subchannels;
  size_t num_subchannels;
};
```

Comment (dgquintas) [inline src/core/client_config/lb_policy_factory.h]: Done.

Comment (ctiller) [inline src/core/client_config/lb_policy_factory.h]: Yeah. Sorry!

**Pull request merged**